vcardtools.py

```diff
@@ -238,7 +266,8 @@ def main():  # pylint: disable=too-many-statements,too-many-branches
                         # save the remaining attributes to the merged vCard
                         vcard_merge = build_vcard(attributes)
                         # write to the file
-                        write_vcard_to_file(vcard_merge, d_path + '.vcard')
+                        write_vcard_to_file(vcard_merge, mk_vcard_fnm(d_path, the_vcard_ext))
```

Comment:
This chunk for *merge* shouldn't be applied since it will sanitize the full path, not just the filename (so */my/path/file?name* will become *_my_path_file_name*). Instead use:
```suggestion
                        write_vcard_to_file(vcard_merge, d_path + the_vcard_ext)
```